sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -71,7 +72,26 @@ class QueryExecution(
 
   lazy val analyzed: LogicalPlan = executePhase(QueryPlanningTracker.ANALYSIS) {
     // We can't clone `logical` here, which will reset the `_analyzed` flag.
-    sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
+    val plan = sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
+
+    // For various commands (like DDL) and queries with side effects, we force query execution
+    // to happen right away to let these side effects take place eagerly.
+    def runCommand(plan: LogicalPlan): LogicalPlan = {
+      val qe = new QueryExecution(sparkSession, plan, isCommand = true)
+      LocalRelation(plan.output,
+        SQLExecution.withNewExecutionId(qe)(qe.executedPlan.executeCollect()),
+        fromCommand = true)
+    }
+
+    if (isCommand) {
+      plan
+    } else {
+      plan.transformUp {
+        case c: Command => runCommand(c)
```

Comment:
Do we need the `isCommand` flag? There is no nested command and it's OK to always find and run command.

Reply:
It's a flag to eager execute the command, otherwise, the logic here would end up into a dead loop and the command would never be executed.